### PR TITLE
docs: add a document describing the architecture of the repository

### PR DESCRIPTION
**Pull request by michaelpj**
Should be self-explanatory.

### CODE REVIEW

File: ARCHITECTURE.adoc [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+Plutus Core is the language that actually goes on the blockchain. Consequently
+this is the absolute core of the codebase, and everything depends on it.
+
+=== `language-plutus-core`
+
+This package implements the Plutus Core language.
+
+This includes:
+
+- Parser
```
Comment (effectfully) [inline ARCHITECTURE.adoc]: Also includes the AST of the language. Also example types and terms.

File: ARCHITECTURE.adoc [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+- Typechecker
```
Comment (effectfully) [inline ARCHITECTURE.adoc]: Various checkers including a type checker.

File: ARCHITECTURE.adoc [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+=== `plutus-wallet-api`
+
+This package should probably be split in two!
+
+The `ledger` sublibrary defines our model of an Extended UTXO ledger, including:
+- The types that
+
+The rest of the package defines the
+
+=== `plutus-emulator`
```
Comment (shmish111) [inline ARCHITECTURE.adoc]: weird formatting here.
```
- The types that

The rest of the package defines the
...
```

File: ARCHITECTURE.adoc [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+This package provides several things:
+
+- The Plutus Tx compiler, which compiles GHC Core into
+Plutus IR.
+- A GHC Core plugin, which is the mechanism by which people use
+the compiler.
+- A couple of special typeclasses which aid with the interface
+between Haskell and Plutus Tx, and has some Template Haskell support for
+generating instances.
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: Delete "has"?

File: ARCHITECTURE.adoc [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+the simulations in the Plutus Playground.
+
+== Contract modelling
+
+We want to help people write "contracts". When we say that, we mean both:
+. A high-level applications that exposes a limited API and
+is suitable for running inside something like a wallet smart contract backend.
+. An implementation of a conceptual "contract" (e.g. a crowdfunding contract),
+including both the on-chain code and the eventual contract application.
+
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: I think the markdown's wrong here.  It renders as "When we say that, we mean both: . A high-level applications that exposes a limited API and is suitable for running inside something like a wallet smart contract backend. . An implementation of a conceptual "contract" (e.g. a crowdfunding contract), including both the on-chain code and the eventual contract application."  Does it need some indentation at the start of lines?

File: ARCHITECTURE.adoc [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+=== `plutus-emulator`
+
+This package defines the chain emulator, which is used for tests, and to back
+the simulations in the Plutus Playground.
+
+== Contract modelling
+
+We want to help people write "contracts". When we say that, we mean both:
+. A high-level applications that exposes a limited API and
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: applications -> application

File: ARCHITECTURE.adoc [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+=== `plutus-playground-lib`
+
+This package contains some library code for the Plutus Playground.
+
+=== `plutus-playground-server`/`marlowe-playground-server`
+
+These packages contain the servers that back the Plutus/Marlowe Playgrounds by
+compiling user code and evaluating their simulations.
+
+They also define executables that generates Purescript bindings for the types that
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: "generates" -> "generate"

File: ARCHITECTURE.adoc [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+the Purescript code needs.
+
+=== `plutus-playground-client`/`marlowe-playground-client`
+
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: / might be interpreted as a subdirectory.

File: ARCHITECTURE.adoc [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+This contains the Plutus/Marlowe Playground client code, written in Purescript.
```
Comment (kwxm) [inline ARCHITECTURE.adoc]: "This contains" -> "These contain"?

File: ARCHITECTURE.adoc (hunk 3 above)
Comment (michaelpj) [reply ARCHITECTURE.adoc]: I just didn't finish writing this section :sweat_smile:

File: ARCHITECTURE.adoc (hunk 5 above)
Comment (michaelpj) [reply ARCHITECTURE.adoc]: This is asciidoc, and it needs a blank line before the start of a list, which I always forget.

**Comment by michaelpj** [comment]
Thanks all.

**Comment by kwxm** [comment]
Could we put links in the architecture document so that readers can go straight to the appropriate pages in the repository?

**Pull request merged**